src/server/pkg/localcache/cache.go

```diff
@@ -0,0 +1,97 @@
+package localcache
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+)
+
+// Cache is a simple unbounded disk cache and is safe for concurrency.
+type Cache struct {
+	root string
+	keys map[string]bool
+	mu   sync.Mutex
```

Comment:
It seems like contention might become a real issue here given that this lock is held while files are copied to and from disk. It could maybe be made better with a `sync.Map`?

Reply:
I looked at that data structure when I was creating this and I am not sure it is what we would need here. Taking into account what I mentioned in the post above on pre-mature optimizations, I think it is unclear if we will fit into the two common use cases they list for that data structure. The case where it only grows is probably not applicable for eviction purposes (particularly when we start putting real data in here). The case where the accesses from multiple go routines is disjoint seems very likely untrue particularly when we would be using this for caching data in crosses. I think a lot of the optimizations we could make here need to be informed by more information as we get to where we need them. After some more testing, I should have a better idea what we really need, and we can continue to iterate on this as we gather more info.

Reply:
Makes sense not to do this optimization until we have some evidence that it's an impactful one.